bookserver/routers/books.py

```diff
@@ -110,19 +111,28 @@ async def serve_page(
     # need to use lowercase true and false.
     if user:
         logged_in = "true"
+        user_is_instructor = await is_instructor(request)
     else:
         logged_in = "false"
-
+        activity_info = {}
     course_row = await fetch_course(user.course_name)
     if not course_row:
         raise HTTPException(status_code=404, detail=f"Course {course_name} not found")
     rslogger.debug(f"Base course = {course_row.base_course}")
+    chapter = os.path.split(os.path.split(pagepath)[0])[1]
+    subchapter = os.path.basename(os.path.splitext(pagepath)[0])
+    if logged_in:
```

Comment:
This is always true. Use `if user:` instead. I'd also suggest putting it in the `if user:` clause above.

Reply:
oops

Reply:
we need course information first, so I put it here

Reply:
Yes, I see that now...